data-plane/receiver/src/test/java/dev/knative/eventingkafkabroker/receiver/integration/ReceiverVerticleTest.java

```diff
@@ -0,0 +1,260 @@
+package dev.knative.eventingkafkabroker.receiver.integration;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.FloatNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import dev.knative.eventingkafkabroker.receiver.CloudEventRequestToRecordMapper;
+import dev.knative.eventingkafkabroker.receiver.HttpVerticle;
+import dev.knative.eventingkafkabroker.receiver.ProducerDriver;
+import dev.knative.eventingkafkabroker.receiver.RequestHandler;
+import io.cloudevents.CloudEvent;
+import io.cloudevents.core.message.Message;
+import io.cloudevents.core.v1.CloudEventBuilder;
+import io.cloudevents.http.vertx.VertxHttpClientRequestMessageVisitor;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpClientRequest;
+import io.vertx.core.http.HttpServerOptions;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import io.vertx.kafka.client.producer.KafkaProducerRecord;
+import io.vertx.kafka.client.producer.impl.KafkaProducerRecordImpl;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(VertxExtension.class)
+public class ReceiverVerticleTest {
+
+  static final int TIMEOUT = 10;
+  static final int PORT = 8080;
+
+  @Test
+  public void requestHandler(final Vertx vertx, final VertxTestContext testContext)
+      throws Throwable {
+
+    final var httpClient = vertx.createHttpClient();
+    final var testCases = requestStream();
+    final var countDown = new CountDownLatch(testCases.size());
+    final var flags = testContext.checkpoint(testCases.size() * 2);
+
+    final var producerDriver = new ProducerDriver<String, CloudEvent>();
+    final var producer = producerDriver.producer(false);
+
+    final var handler = new RequestHandler<>(producer, new CloudEventRequestToRecordMapper());
```

Comment:
How about using the MockProducer in kafka client https://kafka.apache.org/20/javadoc/org/apache/kafka/clients/producer/MockProducer.html more than creating your custom one?

You can wrap it into vertx kafka producer with https://vertx.io/docs/apidocs/io/vertx/kafka/client/producer/KafkaProducer.html#create-io.vertx.core.Vertx-org.apache.kafka.clients.producer.Producer-

Reply:
Let's try to avoid creating mocks when we don't need these

Reply:
Using this:
```java
    final var producerMock = new MockProducer<String, CloudEvent>();
    final KafkaProducer<String, CloudEvent> producer = KafkaProducer.create(vertx, producerMock);
```
I get
```bash
io.vertx.core.VertxException: Connection was closed
```